server/player/player.go

```diff
@@ -587,6 +587,22 @@ func (p *Player) Hurt(dmg float64, src world.DamageSource) (float64, bool) {
 			damageLeft = 0
 		}
 	}
+
+	w, pos := p.World(), p.Position()
+
+	if p.Health()-damageLeft < 0 {
+		hand, offHand := p.HeldItems()
+		if _, ok := hand.Item().(item.Totem); ok {
+			p.applyTotemEffects()
+			p.SetHeldItems(hand.Grow(-1), offHand)
+			return damageLeft, true
+		} else if _, ok := offHand.Item().(item.Totem); ok {
+			p.applyTotemEffects()
+			p.SetHeldItems(hand, offHand.Grow(-1))
+			return damageLeft, true
```

Comment:
These returns are not necessary because the player will not receive invulnerability ticks, will not play damage action, as well as possible damage to armour and retaliatory damage from the "thorns" enchantment.